src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -1,24 +1,178 @@
+  VERIFY_SUBSECTION,
+  AUTH_AUTHENTICATING,
+} from './constants'
+
+// Hook responsible for deriving the authState from localStorage values and
+// providing setters which update the localStorage
+function useAuthState() {
+  const [authState, setAuthState] = useState(AUTH_UNAUTHENTICATED)
+  const [email, setEmail] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  )
+  const [token, setToken] = useState(
+    localStorage.getItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  )
+
+  useEffect(() => {
+    if (!email && !token) {
+      setAuthState(AUTH_UNAUTHENTICATED)
+      return
+    }
+
+    if (email && !token) {
+      setAuthState(AUTH_PREVERIFY)
+      return
+    }
+
+    if (email && token) {
+      setAuthState(AUTH_AUTHENTICATING)
+      isAuthTokenValid(token)
+        .then(v => {
+          setAuthState(AUTH_AUTHENTICATED)
+          return v
+        })
+        .catch(e => {
+          setAuthState(AUTH_AUTHENTICATION_FAILED)
+        })
+    }
+  }, [email, token])
+
+  useEffect(() => {
+    token
+      ? localStorage.setItem(NOTIFICATION_SERVICE_TOKEN_KEY, token)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  }, [token])
+
+  useEffect(() => {
+    email
+      ? localStorage.setItem(NOTIFICATION_SERVICE_EMAIL_KEY, email)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  }, [email])
+
+  const handleLogout = useCallback(() => {
+    setEmail(null)
+    setToken(null)
+  }, [setToken, setEmail])
+
+  return {
+    authState,
+    email,
+    token,
+    handleTokenChange: setToken,
+    handleEmailChange: setEmail,
+    handleLogout,
+  }
+}
+
+export default function Notifications({
+  apps,
+  subsection,
+  dao,
+  handleNavigation,
+  navigationIndex,
+}) {
+  const navigateToNotifications = useCallback(
+    () => handleNavigation(navigationIndex),
+    [handleNavigation, navigationIndex]
   )
+  const {
+    authState,
+    email,
+    token,
+    handleTokenChange,
+    handleEmailChange,
+    handleLogout,
+  } = useAuthState()
+
+  if (subsection && subsection.startsWith(VERIFY_SUBSECTION)) {
+    return (
+      <NotificationsVerify
+        subsection={subsection}
+        onTokenChange={handleTokenChange}
+        onEmailChange={handleEmailChange}
+        navigateToNotifications={navigateToNotifications}
+      />
+    )
+  }
+
+  switch (authState) {
+    case AUTH_PREVERIFY:
+      return <NotificationsPreVerify email={email} />
+    case AUTH_AUTHENTICATED:
+      return (
+        <ManageNotifications
+          dao={dao}
+          onLogout={handleLogout}
+          apps={apps}
+          email={email}
+          token={token}
+        />
+      )
+    case AUTH_UNAUTHENTICATED:
+      return (
+        <NotificationsLogin
+          dao={dao}
+          authState={authState}
+          onEmailChange={handleEmailChange}
+        />
+      )
+    case AUTH_AUTHENTICATING:
+      return (
+        <NotificationsVerifyBox header="Authenticating">
```

Comment:
The containing `grid` does not allow the correct alignment here: 

![image](https://user-images.githubusercontent.com/3072458/63692024-0c6d4e00-c811-11e9-983c-b2ba2e3810f2.png)